src/pybind/mgr/prometheus/module.py

```diff
@@ -322,18 +426,19 @@ def get_metadata_and_osd_status(self):
                 )
                 continue
             dev_class = next((osd for osd in osd_devices if osd['id'] == id_))
-            self.metrics['osd_metadata'].set(1, (
-                c_addr,
-                dev_class['class'],
-                id_,
-                p_addr
-            ))
+            host_version = servers.get(('{}'.format(id_), 'osd'), ('',''))
```

Comment:
Converting the id to a string here is not my favourite. The osd_map contains the OSD id as a long int, the service_map gives us a string. Imho both instances should be the same type. @jcsp I'd be happy to fix this in the mgr code, what is the correct answer here...string or int?

Reply:
Not that it would fix the underlying cause but you can use `str(id_)` here instead.

Reply:
@jan--f I think string is the way to go: the approach in ceph-mgr is to make all IDs strings.  While the OSDs have integer IDs, everything else (mon, mds, etc) uses strings, so to avoid mixing the types the OSD ids are cast to strings too.